include/bluetooth/conn.h

```diff
@@ -800,6 +800,16 @@ struct bt_conn_cb {
 	 */
 	void (*disconnected)(struct bt_conn *conn, uint8_t reason);
 
+	/** @brief The connection MTU has been updated.
+	 *
+	 *  This callback notifies the application that a remote device
+	 *  has changed the connection MTU.
+	 *
+	 *  @param conn Connection object.
+	 *  @param mtu Updated MTU.
+	 */
+	void (*mtu_updated)(struct bt_conn *conn, uint16_t mtu);
```

Comment:
I see the value in adding this callback. Especially because this would make it easier when you have multiple GATT profiles and you want something to happen once the MTU has been updated.

This is related to the fact that the MTU exchange should only be initiated once, and multiple profiles might attempt to do just that. At the moment we don't enforce this rule.

 I would therefore like to extend this so that it also called when it has been initiated locally, i.e as a GATT client.
 Also we should make sure it is only called once per connection, that would make it easier for applications, since the ATT mtu should end up as the same in both procedures (as server and client).
 
 Also the name should be updated to reflect that it is the ATT MTU that has changed. MTU of for example L2CAP Channels are handled differently.

Reply:
> This is related to the fact that the MTU exchange should only be initiated once

Is this part of the specification? I was happily updating the MTU multiple times per connection through the nRF Connect app when testing this PR

Reply:
> Also the name should be updated to reflect that it is the ATT MTU that has changed

The naming was chosen to be consistent with `bt_conn_get_mtu`. I 100% agree that it is not clear. Is there more value in being consistent or in making the new function clearer?

Reply:
> The naming was chosen to be consistent with bt_conn_get_mtu. I 100% agree that it is not clear. Is there more value in being consistent or in making the new function clearer?

Let's get some more opinions on this before we make this decision.

> Is this part of the specification? I was happily updating the MTU multiple times per connection through the nRF Connect app when testing this PR

Yes, it is part of the specification, but the API does not enforce this rule at the moment.

Reply:
> > This is related to the fact that the MTU exchange should only be initiated once
> 
> Is this part of the specification? I was happily updating the MTU multiple times per connection through the nRF Connect app when testing this PR

It has a recommendation that MTU exchange is only run once, anyway most of the time having another exchanges makes no different as the values won't change.

Reply:
@Vudentz There is no recommendation, it is a rule.
"This request shall only be sent once during a connection by the client"